File: backend/src/main/java/com/skillswap/entity/Notification.java
package com.skillswap.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "notifications")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Notification {

    @Id
    @GeneratedValue
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user; // recipient

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private NotificationType type;

    @Column(nullable = false, length = 500)
    private String message;

    // Optional session this notification relates to, so the frontend can
    // deep-link straight to it.
    private UUID relatedSessionId;

    @Column(nullable = false)
    @Builder.Default
    private boolean isRead = false;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private Instant createdAt;
}

File: frontend/src/types/notification.ts
export type NotificationType =
  | "SESSION_REQUESTED"
  | "SESSION_ACCEPTED"
  | "SESSION_REJECTED"
  | "SESSION_COMPLETED"
  | "REVIEW_RECEIVED";

export interface AppNotification {
  id: string;
  type: NotificationType;
  message: string;
  relatedSessionId: string | null;
  isRead: boolean;
  createdAt: string;
}
